doc/memory_statistics/memory_statistics_hld.md

```diff
@@ -0,0 +1,373 @@
+
+	admin@sonic:~$ config memory-stats retention-period <period>
+ Default retention-period is 15 days
+
+**Enable/Disable Memory Statistics Monitoring**
+
+To enable or disable the memory statistics monitoring feature, use the following command:
+
+	admin@sonic:~$ config memory-stats enable/disable
+ By default, it is disabled.
+
+**View Memory Usage**
+
+To display memory usage statistics, use the following command with optional parameters for time range and specific metrics:
+
+	admin@sonic:~$ show memory-stats [--from <date-time>] [--to <date-time>] [--select <metric>]
+
+**Command Definition**
+  - **show memory-stats:** Display basic memory usage statistics
+  - **--from <date-time>:** Display memory statistics from the specified start date-time.
+  - **--to <date-time>:** Display memory statistics up to the specified end date-time.
+  - **--select <metric>:** Display specific memory statistics, such as total memory.
+
+ **Sample Output for Memory Usage**
+
+Below is an example of the Memory Statistics output as it appears in the CLI. This display provides a summary of system memory metrics over a default time period, ideal for routine monitoring and analysis:
+
+
+    admin@sonic:~$ show memory-stats 
+
+
+    Codes:	M - minutes, H - hours, D - days
+	--------------------------------------------------------------------------------
+	Report Generated:    2024-06-15 09:00:00
+    Analysis Period:     From 2024-06-01 09:00:00 to 2024-06-15 09:00:00
+    Interval:	         2 days
+	--------------------------------------------------------------------------------
+	  Metric            Current     High       Low        D1-D3      D3-D5      D5-D7      D7-D9      D9-D11     D11-D13    D13-D15    
+	                    Value       Value      Value      01Jun24    03Jun24    05Jun24    07Jun24    09Jun24    11Jun24    13Jun24    
+	--------------      --------    --------   --------   ---------  ---------  ---------  ---------  ---------  ---------  ---------
+	Total Memory        15.6G       15.6G      15.1G      15.1G      15.2G      15.3G      15.3G      15.4G      15.5G      15.5G      
+	Used Memory         2.3G        2.5G       2.0G       2.1G       2.2G       2.2G       2.3G       2.4G       2.3G       2.2G       
+	Free Memory         11.9G       12.4G      10.6G      11.0G      11.2G      11.4G      11.5G      11.7G      11.8G      11.9G      
+	Available Memory    13.0G       14.3G      12.4G      12.6G      12.7G      12.8G      12.9G      13.0G      13.1G      13.2G      
+    Cached Memory       1.2G        1.5G       1.0G       1.1G       1.2G       1.3G       1.4G       1.3G       1.4G       1.2G       
+    Buffer Memory       0.3G        0.4G       0.2G       0.2G       0.3G       0.3G       0.4G       0.3G       0.3G       0.4G       
+	Shared Memory       0.5G        0.6G       0.4G       0.5G       0.5G       0.5G       0.4G       0.5G       0.5G       0.5G       
+
```

Comment:
@Arham-Nasirare these stats collected from the output of `free` linux command?

Reply:
The data is not directly collected from the output of the free Linux command. Instead, they are gathered using the psutil Python library, which directly accesses system memory information via the same underlying mechanisms that the free command relies on. Specifically, the psutil.virtual_memory() function retrieves memory stats such as total, used, free, buffers, cached, and available memory.This approach allows for more direct access to memory data, avoiding the need to parse command output while providing flexibility for programmatic use.